app/javascript/components/redfish-server-firmware-update-dialog.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import FirmwareUpdate from "./forms/firmware-update"
+import { handleApiErrorHooks, fetchFirmwareBinariesForServer, createFirmwareUpdateRequest } from "../utils/api";
+import { selectedPhysicalServers } from "../utils/common";
+
+const RedfishServerFirmwareUpdateDialog = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [firmwareBinaries, setFirmwareBinaries] = useState([]);
+  const [error, setError] = useState('');
+  const [values, setValues] = useState({});
+  const physicalServerIds = selectedPhysicalServers();
+
+  useEffect(() => {
+    props.dispatch({
+      type: "FormButtons.init",
+      payload: {
+        newRecord: true,
+        pristine: true,
+      }
+    });
+    props.dispatch({
+      type: "FormButtons.customLabel",
+      payload: __('Apply Firmware')
+    });
+    fetchBinaries()
+  }, []);
+
+  // Everytime a dropdown value changes we have to update the callback function so that its
+  // closure contains the most recent state values.
+  useEffect(() => {
+    props.dispatch({
+      type: "FormButtons.callbacks",
+      payload: {
+        addClicked: () => createFirmwareUpdateRequest(physicalServerIds, values.firmwareBinary)
+      }
+    });
+  }, [values]);
+
+  const fwBinaryToSelectOption = fwBinary => { return { value: fwBinary.id, label: `${fwBinary.name} (${fwBinary.description})` } };
+
+  const fetchBinaries = () => fetchFirmwareBinariesForServer(physicalServerIds[0]).then((fwBinaries) => {
```

Comment:
Is it possible for `physicalServerIds[0]` to be empty?

Reply:
No - the popup is not supposed to be visible unless we pick at least one server. Thanks for pointing out, though, added an IF into the useEffect so that the component will print a nice error message.